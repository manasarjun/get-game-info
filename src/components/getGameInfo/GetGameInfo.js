import React, { useRef, useState } from 'react';
import GameSchedule from "../GameSchedule/GameSchedule";
import fetchData from '../../utils/fetchData'
import '../../App.css';

function GetGameInfo() {
  const gameType = useRef(null);
  const [closestGame, setClosestGame] = useState();
  const [displayGameType, setdisplayGameType] = useState(null);
  const handleError = (err) => {
    alert('Please enter a valid game type (Valid Game Types: V4, V64, V65, V75)')
    console.log('in handler---->', err);
  }
  const handleOnSubmit = (e) => {
    e.preventDefault();
    const keyword = gameType.current.value.toUpperCase();
    fetchData(keyword)
      .then(res => {
        if (res.status !== 200 && !res.ok) {
          throw new Error(res);
        }
        return res.json();
      })
      .then(data => {
        setdisplayGameType(data.betType);
        data.upcoming ? setClosestGame(data.upcoming[0]) : setClosestGame(data.results[0])
      }).catch(handleError)
    gameType.current.value = '';
  }

  return (
    <>
      <form onSubmit={handleOnSubmit}>
        <input
          className='text-field'
          type='text'
          placeholder='enter game type'
          ref={gameType}
        />
      </form>
      {displayGameType ? <h1>Game Type: {displayGameType}</h1>
        : <p>Please enter a game type</p>
      }
      {closestGame && <GameSchedule game={closestGame} />}
    </>
  )
}

export default GetGameInfo;

