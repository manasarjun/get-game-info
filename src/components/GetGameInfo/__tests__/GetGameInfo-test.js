import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import GetGameInfo from '../GetGameInfo'

describe('<GetGameInfo />', () => {
  it('Render Initial Form', () => {
    const wrapper = shallow(<GetGameInfo />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('input')).toHaveLength(1)
    expect(wrapper.find('h1')).toHaveLength(0)
  });
});
