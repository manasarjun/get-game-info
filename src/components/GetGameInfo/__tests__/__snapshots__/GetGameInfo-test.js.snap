// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetGameInfo /> Render Initial Form 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      className="text-field"
      placeholder="enter game type"
      type="text"
    />
  </form>
  <p>
    Please enter a game type
  </p>
</Fragment>
`;
